File: backend/schemas/approval.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from models.enums import ApprovalStatus, ApprovalType, RiskLevel


class ApprovalCreate(BaseModel):
    workflow_id: UUID
    approval_type: ApprovalType
    requested_by: UUID | None = None
    expires_at: datetime | None = None
    risk_level: RiskLevel = RiskLevel.low


class ApprovalResolve(BaseModel):
    approval_status: ApprovalStatus  # approved or rejected
    resolved_by: UUID
    resolution_notes: str | None = None


class ApprovalResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    workflow_id: UUID
    approval_type: ApprovalType
    approval_status: ApprovalStatus
    requested_by: UUID | None
    resolved_by: UUID | None
    expires_at: datetime | None
    risk_level: RiskLevel
    resolution_notes: str | None
    created_at: datetime
    updated_at: datetime

File: frontend/types/approval.ts
// Mirrors backend ApprovalType enum
export type ApprovalType =
  | 'offer'
  | 'price_reduction'
  | 'skip_trace'
  | 'outreach'

// Mirrors backend ApprovalStatus enum
export type ApprovalStatus =
  | 'pending'
  | 'approved'
  | 'rejected'
  | 'expired'
  | 'escalated'

// Mirrors backend RiskLevel enum
export type RiskLevel = 'low' | 'medium' | 'high' | 'critical'

// Mirrors backend ApprovalResponse schema
export interface Approval {
  id: string
  workflow_id: string
  approval_type: ApprovalType
  approval_status: ApprovalStatus
  requested_by: string | null
  resolved_by: string | null
  expires_at: string | null
  risk_level: RiskLevel
  resolution_notes: string | null
  created_at: string
  updated_at: string
}

export interface ApprovalPage {
  items: Approval[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface ApprovalCreateBody {
  workflow_id: string
  approval_type: ApprovalType
  requested_by?: string | null
  risk_level?: RiskLevel
  expires_at?: string | null
}

export interface ApprovalResolveBody {
  approval_status: 'approved' | 'rejected'
  resolved_by: string
  resolution_notes?: string | null
}
